ibis/expr/types/arrays.py

```diff
@@ -153,71 +155,59 @@ def __add__(self, other: ArrayValue) -> ArrayValue:
         │ [3]                  │
         │ NULL                 │
         └──────────────────────┘
-        >>> t.a + t.a
+        >>> t.a.concat(t.a)
         ┏━━━━━━━━━━━━━━━━━━━━━━┓
-        ┃ ArrayConcat(a, a)    ┃
+        ┃ ArrayConcat()        ┃
         ┡━━━━━━━━━━━━━━━━━━━━━━┩
         │ array<int64>         │
         ├──────────────────────┤
         │ [7, 7]               │
         │ [3, 3]               │
         │ NULL                 │
         └──────────────────────┘
-        >>> t.a + ibis.literal([4], type="array<int64>")
+        >>> t.a.concat(ibis.literal([4], type="array<int64>"))
         ┏━━━━━━━━━━━━━━━━━━━━━━┓
-        ┃ ArrayConcat(a, (4,)) ┃
+        ┃ ArrayConcat()        ┃
         ┡━━━━━━━━━━━━━━━━━━━━━━┩
         │ array<int64>         │
         ├──────────────────────┤
         │ [7, 4]               │
         │ [3, 4]               │
         │ [4]                  │
         └──────────────────────┘
-        """
-        return ops.ArrayConcat(self, other).to_expr()
-
-    def __radd__(self, other: ArrayValue) -> ArrayValue:
-        """Concatenate this array with another.
 
-        Parameters
-        ----------
-        other
-            Array to concat with `self`
+        `concat` is also available using the `+` operator
 
-        Returns
-        -------
-        ArrayValue
-            `self` concatenated with `other`
-
-        Examples
-        --------
-        >>> import ibis
-        >>> ibis.options.interactive = True
-        >>> t = ibis.memtable({"a": [[7], [3] , None]})
-        >>> t
+        >>> [1] + t.a
         ┏━━━━━━━━━━━━━━━━━━━━━━┓
-        ┃ a                    ┃
+        ┃ ArrayConcat()        ┃
         ┡━━━━━━━━━━━━━━━━━━━━━━┩
         │ array<int64>         │
         ├──────────────────────┤
-        │ [7]                  │
-        │ [3]                  │
-        │ NULL                 │
+        │ [1, 7]               │
+        │ [1, 3]               │
+        │ [1]                  │
         └──────────────────────┘
-        >>> ibis.literal([4], type="array<int64>") + t.a
+        >>> t.a + [1]
         ┏━━━━━━━━━━━━━━━━━━━━━━┓
-        ┃ ArrayConcat((4,), a) ┃
+        ┃ ArrayConcat()        ┃
         ┡━━━━━━━━━━━━━━━━━━━━━━┩
         │ array<int64>         │
         ├──────────────────────┤
-        │ [4, 7]               │
-        │ [4, 3]               │
-        │ [4]                  │
+        │ [7, 1]               │
+        │ [3, 1]               │
+        │ [1]                  │
         └──────────────────────┘
         """
-        return ops.ArrayConcat(other, self).to_expr()
+        return ops.ArrayConcat((self, other, *args)).to_expr()
+
+    def __add__(self, other: ArrayValue) -> ArrayValue:
+        return self.concat(other)
+
+    def __radd__(self, other: ArrayValue) -> ArrayValue:
+        return ops.ArrayConcat((other, self)).to_expr()
```

Comment:
`other.concat(self)`?

Reply:
`other` can be a different type than `ArrayValue`. I should probably remove or fix the type annotation there.

Reply:
Right, and `self` cannot be used because of the concat order.

Reply:
Regarding the annotations we should have type aliases at some point, e.g. `ArrayLike` which contains the types coercible to array values as well.